pkg/channel/fanout/fanout_event_handler.go

```diff
@@ -220,13 +219,35 @@ func createEventReceiverFunction(f *FanoutEventHandler) func(context.Context, ch
 			reportArgs.EventType = evnt.Type()
 			reportArgs.Ns = ref.Namespace
 
+			var (
+				httpCount  int = 0
+				httpsCount int = 0
+			)
+
+			for _, sub := range subs {
+				if sub.Subscriber.URL != nil && sub.Subscriber.URL.Scheme == "https" {
+					httpsCount++
+				} else {
+					httpCount++
+				}
+			}
+
+			if httpsCount > 0 {
+				reportArgs.EventScheme = "https"
+			} else {
+				reportArgs.EventScheme = "http"
+			}
 			go func(e event.Event, h nethttp.Header, s *trace.Span, r *channel.StatsReporter, args *channel.ReportArgs) {
 				// Run async dispatch with background context.
 				ctx = trace.NewContext(context.Background(), s)
 				h.Set(apis.KnNamespaceHeader, ref.Namespace)
 				// Any returned error is already logged in f.dispatch().
 				dispatchResultForFanout := f.dispatch(ctx, subs, e, h)
 				_ = ParseDispatchResultAndReportMetrics(dispatchResultForFanout, *r, *args)
+				if httpCount > 0 && httpsCount > 0 {
```

Comment:
It would be helpful to add a comment here explaining why we are reporting the metrics a second time in this case, and why we chose http for this case